### ISSUE TITLE
Add possibility to define custom blocked services

### ISSUE THREAD

**Issue by alexruf** [AdguardTeam/AdGuardHome#1692]
The feature just quickly block & unblock popular sites and services in the general tab with a simple click is awesome.
As a father of children who sometimes overdo it with online gaming, it would be great if you could define your own custom services there to extend and customize the list of services.
I think of it like this:
You simply enter the name of your service and it creates a small separate block list associated with this service name. With a toggle button somewhere in the UI you now can simply enable enable and disable the blocking of these DNS entries.

In theory you could go even further and also add the ability to schedule times when those services are automatically blocked.
I think this could be a really useful feature.

**Comment by Archive5** [comment]
>In theory you could go even further and also add the ability to schedule times when those services are automatically blocked.

#1203 #951

@ameshkov this feature had requested by someone past and closed due certain reason. Opened issue exist for #1605 too as i remember.
I forgot to say about this problem past when was my old account blocked. I had to beg them week for unblock temporally  (usually small reason against them or big brother enough rather than unacceptable or illegal things) 

Below two suggestions to use for adguard's closed source applications atleast :)

- https://getfider.com/ open source 
interface: https://feedback.fider.io
(will available better themes, may be add ways to sync issues, comments between github etc.)
- https://www.zoho.com/desk/ paid
interface: https://desk.zoho.com/portal/zeroalt/community
https://help.zoho.com/portal/en/community
(it's perfect but hard to hope such issue sync feature as above neither fider implement that easily)

**Comment by szolin** [comment]
Is it really necessary to have a custom list of blocked services?

1. We have a compiled static list for the most popular services - it's much better to add more services there, because it will automatically help other people.  Plus our community will keep this list up to date.
2. If there are several domains you want to block for specific clients - you can use `$ctag` filters for that.  I agree it's not *that* simple as click on enable/disable button, but it will do exactly the same.

**Comment by Archive5** [comment]
Well, +1 vote for this feature and due #1721 @eoopx

It's good to add explanation for 'blocked services' creation wizard. below some basic.

- necessary to explore all domains belongs to particular service and add otherwise it may not get blocked properly
- necessary to find all domains related desktop (windows, linux etc), mobile (android, iOS etc), consoles (xbox, nvidea), equipment (switches, smart devices) and hardware (alexa, google home) products otherwise some are may remain accessible
- necessary to check them regularly and update (eg, per month) to prevent occur unwanted accesses.

**Comment by ameshkov** [comment]
Let me postpone this task a little bit, we have too much on v0.103 already.

Regarding defining custom services, let's discuss.

1. Keep the list as is and do nothing
2. Allow people to define their custom sets (kinda overlaps with filter lists)
3. Allow people to pull request more "services" to AdGuard Home. We may need to change the UI a little bit because the list of services will grow huge

**Comment by valaria247** [comment]
> 2. Allow people to define their custom sets (kinda overlaps with filter lists)

Fine. let people to add their own [2] and this feature can be improve more than current.

**Comment by dark0mode** [comment]
options: 2

**Comment by cl0ud-o** [comment]
2

**Comment by esvra** [comment]
2

**Comment by valaria247** [comment]
> Let me postpone this task a little bit, we have too much on v0.103 already.

@ameshkov last time this postponed to v0.104 but secretly has moved to v0.106
No much to-do list on v0.104 so can you add this back to it?

**Comment by ameshkov** [comment]
We're going to conduct some massive refactoring in v0.104 or v0.105 so I am keeping these releases without new features.

**Comment by valaria247** [comment]
> We're going to conduct some massive refactoring in v0.104 or v0.105 so I am keeping these releases without new features.

Okey :) If there any chance later then consider to assign to **v0.104*** or 5

**Comment by ameshkov** [comment]
Zero chances

**Comment by ameshkov** [comment]
As I said, I am keeping 0.104/0.105 without new features. We need this refactoring to integrate a more important feature -- content blocking proxy. Unfortunately, everything else will have to wait.

**Comment by valaria247** [comment]
> Zero chances

(๑◕︵◕๑)

**Comment by Aikatsui** [comment]
Regarding @ameshkov https://github.com/AdguardTeam/AdGuardHome/issues/1692#issuecomment-634512195
2. Allow people to define their custom sets. (kinda overlaps with filter lists)
 ~~ Why? It's nice to have them in(as) tiles/icons (instead lists format) to toggle when necessary  rather than add and  struggle on "blocklists" page with existing ad-block/tracking blocklists and we can't ask to add any interested services according to our desires. 

(not related to above)
I add comment on this issue because contain on near milestone. @ameshkov @ainar-g 
v0.107 (The purpose of this details to make it an one control panel, redundancy and less complicated as some people's point of view. Both methods doesn't affect to the product facilities. If existing method seems okey then ofcourse you can ignore this comment)

Even this part separated as "blocked services with icons" but still same as block-lists available on "DNS blocklists" page then this issue is about add custom list(service)

There're two issues about "Schedule"
1. Add possibility to setup schedule for blocked services (and more) on per-client basis - https://github.com/AdguardTeam/AdGuardHome/issues/951 v0.109
2. Timed blocklists and whitelists - https://github.com/AdguardTeam/AdGuardHome/issues/1203 v0.108
(Different client filter configuration https://github.com/AdguardTeam/AdGuardHome/issues/435 may get added eventually)

Both issues has closer amount of votes and If you implemented only 1st then users may over-use that feature for schedule anything they prefer but it's like you've developed something incomplete to cover users needs and if you only implemented 2nd then some users may still looking it for "blocked services" page However i see 2nd option can be used for various purposes, might helpful to attach new flexible features and far more extendable with client settings, user accounts #997 etc.

None of that, i prefer suggest to combine that all 3 issues and 2 existing components

**2.Timed blocklists also add to this same page (DNS blocklists). 
(another feature: add sort option to view block-lists under category https://github.com/AdguardTeam/AdGuardHome/issues/2322)

- Merge "Blocked Services" page to top of "DNS blocklists" page. To manage the page space then only display 3 or 4 lines(rows) services icons and let to expand the rest manually (users dont enable/disable them often)
- Single setup for all "add blocklist or service". Only one wizard necessary to create ad/tracking blocklists, custom blocklists, custom services or timed blocklists
- Single "Edit" window for all. Excluding pre-defined ad-block lists. User can convert any list to service and vise-versa or change any lists or service to be scheduled.
- Condition for blocked services 
    - If user selected to create list as service or edited existing list for change as service then it'll display on the top as icon/tiles
    - If user configured any service to be scheduled then it must disappear from top tiles and appear in timed blocklist section as lists format. (Then tiles/icons format only remain if it just keep as a simple service to enable and disable) 

Just then only a one feature, no overlaps and trick user through front UI to cover the requirement

About loss blocked services page and you have something new to replace it
Solution to block worldwide services under particular categories #2230
(That available on comodo domeshield, cleanbrowsing, nextdns and other platforms)

Collapsible Sections #1409

![ksnip_20201119-032012](https://user-images.githubusercontent.com/63988538/99601292-b4cc8b00-2a31-11eb-9312-87a7ceba61bc.png)
(some image contents taken from other issues)

![Th8Y0vn](https://user-images.githubusercontent.com/63988538/99601963-19d4b080-2a33-11eb-828c-0246c66a9ee2.png)

**Comment by tomcoleman** [comment]
alexruf that would be amazing what you posted!!!

**Comment by vipinjn24** [comment]
Anything on this ticket going on?

**Comment by alexsannikov** [comment]
Bump this ticket. This feature request still looks very attractive, and is still awaiting. The main idea of this suggestion is the very quick allow/disallow some customers pre-defined service. It may be not very well known, or maybe even very narrowly user-specific, and never to be included into the new release. 
Of course, adding and removing the domain (one or many) into custom rules does the same, but one toggle is much more useful and simple. Thanks

**Comment by overwatch3560** [comment]
@alexruf is your issue resolved by now?

**Comment by quaintdev** [comment]
Currently, time schedule-based blocking can only be applied to services listed in the Blocked Services. This limitation prevents users from applying time-based restrictions on any services not included in this list. Expanding this feature would allow for greater flexibility and control, accommodating the vast array of regional and global services that users may want to restrict based on specific schedules. This enhancement is highly needed to provide comprehensive time-based blocking capabilities.

**Comment by rsurgiewicz** [comment]
Hi Folks,
Any updates about this fancy feature?
I'd really love to see that Custom blocked services, as some of the filters does not work.

E.g., blocking reddit does not block `redd.it` nor similar

<img width="2598" height="2014" alt="Image" src="https://github.com/user-attachments/assets/bf854508-e82a-4fcd-b286-5040b5ff6c2b" />
